frontend/src/devicelist/ContentsTable.js

```
@@ -0,0 +1,68 @@
+import React from 'react';
+import { Box } from "@material-ui/core"
+
+import * as DeviceApi from "../api/DeviceApi";
+import GenerateData from './SampleData'
+import SearchBar from './SearchBar'
+import SortBy from './SortBy'
+import VerticalTabs from './VerticalTabs'
+import TabPanel from './TabPanel'
+import DevicesTable from './DevicesTable'
+
+export default class ContentsTable extends React.Component {
+    constructor(props) {
+        super(props)
+        var sampleData = GenerateData();
+        this.state = {
+            senders: sampleData.senders,
+            receivers: sampleData.receivers,
```

I think we can remove this sampleData now that you have componentDidMount() handling the data loading. Or we could  use the DeviceApi here two. What I'm getting at is that we should use the same source for the sample data and "channel" to load it even if it's temporary data.

---

@felixlapierre ^

---

Addressed in 0c5f8aac, we now generate sample data when we fail to connect to the server, instead of loading it in by default and then replacing it. I do want to keep the sample data for now though since it lets us visualize all the possible values in the UI easily.